src/depthcloud_encoder.cpp

```diff
@@ -77,12 +77,18 @@ DepthCloudEncoder::DepthCloudEncoder(ros::NodeHandle& nh, ros::NodeHandle& pnh)
   // read point cloud topic from param server
   priv_nh_.param<std::string>("cloud", cloud_topic_, "");
 
+  // read camera info topic from param server.
```

Comment:
Not from the param server.

Reply:
It is, from the launch file parameter.

Reply:
Instead of reading the topic **name** from the param server, why not use topic remapping?

Reply:
What is the benefit?

Reply:
+1 for remapping. that's the ros way

Reply:
The `cloud` topic is also provided the same way. If there are no clear benefits of one over the other, I prefer to wrap this up. Thanks for the information.

Reply:
Mainly because it's idiomatic. Part of my initial confusion was due to this. I would never use a param to specify a topic name, I would use [arg remapping](http://wiki.ros.org/Remapping%20Arguments), which is meant for that.

Arg remapping is also explicit in a launch file, and helps you to understand how to plug a node into a new sytem. It's a nice way to see the node's interaction with the outside world, which is more obfuscated when you use the rosparam alternative.

Then there's the fact that we don't pollute the param server with a value that is only used once at launch time.

Reply:
> The `cloud` topic is also provided the same way. If there are no clear benefits of one over the other, I prefer to wrap this up. Thanks for the information.

I don't see any benefit of the param approach other than that it doesn't require making changes to the current state of the PR.